src/Common/Commands.Common.Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -62,7 +64,26 @@ public AuthenticationFactory()
                 configuration.ClientRedirectUri,
                 configuration.ResourceClientUri,
                 configuration.ValidateAuthority);
-            if (account != null && environment != null
+            if (account != null && account.Type == AzureAccount.AccountType.ManagedService)
+            {
+                if (environment == null)
+                {
+                    throw new InvalidOperationException("Environment is required for MSI Login");
+                }
+
+                if (!account.IsPropertySet(AzureAccount.Property.MSILoginUri))
+                {
+                    account.SetProperty(AzureAccount.Property.MSILoginUri, DefaultMSILoginUri);
+                }
+
+                if (string.IsNullOrWhiteSpace(tenant))
+                {
+                    tenant = environment.AdTenant?? "Common";
```

Comment:
Nit: the [null-coalesce operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-conditional-operator) is typically spaced on both sides like the ternary (`?:`) operator.